### PR TITLE
Fix upgrading pip/setuptools/wheel during AppVeyor CI builds

### PR DESCRIPTION
**Pull request by frerich**
The CI builds started failing a while ago, printing:

  pip install -U pip setuptools wheel
  ERROR: To modify pip, please run the following command:
  c:\python34\python.exe -m pip install -U pip setuptools wheel

I.e. it sounds as if one should not invoke 'pip' directly for some
reason but rather load it as a Python module. Looking at the YAML file,
it seems we already did that -- right after the problematic 'pip install
-U' call! So let's drop the direct 'pip' call in favor of the existing
'python -m pip ...' call to fix this.

### CODE REVIEW

**Comment by formiaczek** [comment]
ouh, I didn't check other PRs.. and tried to addressed the same :) I think the complaint was about the pip installing on top of itself.. and in 2cc04bb  I thought that this pip install was not really needed after all, invoking it through the module will update pip itself without complaints :) I'll drop this commit / rebase once you've merged your fix :)

File: clcache/storage.py [hunk 1]
```diff
@@ -156,7 +156,7 @@ def clean(self, stats, maximumSize):
                                 maximumSize)
 
 
-class CacheFileWithMemcacheFallbackStrategy(object):
```
Comment (formiaczek) [inline clcache/storage.py]: wasn't worth keeping it, given you've disabled the warning in the other commit anyway? I guess apart from one line - the whole codebase was is still compatible with python 2.7 - but I guess on Windows it's never the case that there needs to be python 2.7 ?

**Comment by formiaczek** [comment]
thanks for fixing this :+1:

**Comment by frerich** [comment]
Alas, I don't understand why the `coverage` program can be executed as a plain shell command in the `.yml` file, but not in a PowerScript script. :-(

File: appveyor.yml [hunk 2]
```diff
@@ -96,7 +97,7 @@ test_script:
       $wc = New-Object 'System.Net.WebClient'
       $wc.UploadFile("https://ci.appveyor.com/api/testresults/junit/$($env:APPVEYOR_JOB_ID)", (Resolve-Path .\unittests.xml))
 
-      & codecov --no-color -X gcov -F unittests -e PYTHON_INSTALL
+      & $($env:PYTHON_INSTALL)\Scripts\codecov.exe --no-color -X gcov -F unittests -e PYTHON_INSTALL
```
Comment (izmmisha) [inline appveyor.yml]: Probably it should looks like `${env:PYTHON_INSTALL}` instead `$($env:PYTHON_INSTALL)`
Also it should be changed at the line 117 
```
 & codecov --no-color -X gcov -F integrationtests_memcached -e PYTHON_INSTALL
```

Comment (frerich) [reply appveyor.yml]: I'll try that (I took the syntax from the line right above, where the URL to upload the results to is constructed)!

Comment (frerich) [reply appveyor.yml]: I also don't understand why it doesn't just pick up the `codecov` executable which is reachable in the` PATH` -- maybe the PowerShell script does not use the `PATH` the way I expect it to...

Comment (izmmisha) [reply appveyor.yml]: Ok, I'm able to reproduce this error on local machine. Problem is related to codecov:
```
==> Preparing upload
    Error running `cd 'C:\projects\clcache' && git ls-files`: None
```
It try to `Popen("cd 'C:\projects\clcache' && git ls-files", stdout=PIPE, shell=True)` and it fails with 'The filename, directory name, or volume label syntax is incorrect.'
Here is the code:
```
try:
    from shlex import quote
except ImportError: # pragma: no cover
from pipes import quote
[...]
 root = quote(root)
[...]
 toc = str((try_to_run('cd %s && git ls-files' % root) or
[...]
```
quote's result is `"'C:\\projects\\clcache'"`

**Comment by izmmisha** [comment]
Ok, one more problems I discovered:
```
Codecov v2.0.15
==> Detecting CI provider
    AppVeyor Detected
    Fixing merge commit SHA
==> Preparing upload
    Error running `cd 'C:\projects\clcache' && git ls-files`: None
XX> Skip processing gcov
==> Collecting reports
XX> Searching for reports disabled
Error: No coverage report found
```
codecov can't find coverage report if `coverage.xml` is already exists, so need to remove `coverage xml` call, codecov will call it.

`codecov.exe : The filename, directory name, or volume label syntax is incorrect.` this exception probably can be avoided if redirect codecov stderr output to null (`2> $null`), but not sure if this helps.

**Comment by izmmisha** [comment]
Well, maybe better would to move from codecov-python to codecov-exe, I'm tried it - results at 
https://ci.appveyor.com/project/izmmisha/clcache/build/1.0.3/job/qahj29815ccvi2jc

**Comment by izmmisha** [comment]
@frerich the problem with
```
 codecov.exe : The filename, directory name, or volume label syntax is incorrect.
    At line:9 char:1
    + & codecov --no-color -X gcov -F unittests -e PYTHON_INSTALL
    + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        + CategoryInfo          : NotSpecified: (The filename, d...x is incorrect.:String) [], RemoteException
        + FullyQualifiedErrorId : NativeCommandError
```
is related to codecov itself, here is my [comment](https://github.com/frerich/clcache/pull/321#discussion_r206811310) and issue https://github.com/codecov/codecov-python/issues/156

invocation of codecov is correct.

**Comment by izmmisha** [comment]
Coverage report still not uploading via codecov: `Error: No coverage report found`.
according to the [code](https://github.com/codecov/codecov-python/blob/master/codecov/__init__.py#L675) 
```
            if os.path.exists(opj(os.getcwd(), '.coverage')) and not os.path.exists(opj(os.getcwd(), 'coverage.xml')):
                write('    Generating coverage xml reports for Python')
                # using `-i` to ignore "No source for code" error
                try_to_run('coverage xml -i')
                reports.append(read(opj(os.getcwd(), 'coverage.xml')))
```
`coverage.xml` will be append to reports if `coverage.xml` not exists and codecov will generate it by `coverage xml -i`

**Pull request merged**